### PR TITLE
Stable Diffusion with OV TorchDynamo backend

### PR DESCRIPTION
**Pull request by likholat**
This notebook demonstrates how to use the `stable-diffusion-2-1-base` model with [OpenVINO TorchDynamo backend](https://docs.openvino.ai/2023.1/pytorch_2_0_torch_compile.html)

### CODE REVIEW

**Comment by likholat** [comment]
@ynimmaga @devangaggarwal

File: notebooks/257-stable-diffusion-torchdynamo-backend/257-stable-diffusion-torchdynamo-backend.ipynb [hunk 1]
```diff
@@ -0,0 +1,302 @@
+{
```
Comment (eaidova) [inline notebooks/257-stable-diffusion-torchdynamo-backend/257-stable-diffusion-torchdynamo-backend.ipynb:1]: <p>please add table of content</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (eaidova) [inline notebooks/257-stable-diffusion-torchdynamo-backend/257-stable-diffusion-torchdynamo-backend.ipynb:1]: <p>actually not necessary that order of GPU will be this one. it looks like some misleading info. any system with multiple gpu will have indexes (e.g. if I have not intel GPU, but have 3 cuda gpu, they also will be listed as GPU.0, GPU.1, GPU.2).</p><p> </p><p>We use widgets for selecting devices in notebooks please based on openvino.Core.available_devices info. I think it makes sense to use the same approach here too. Please check other notebooks as examples</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (eaidova) [inline notebooks/257-stable-diffusion-torchdynamo-backend/257-stable-diffusion-torchdynamo-backend.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    import openvino.torch  # noqa: F401
</pre><p>I think it is necessary say that for activating openvino backend for torchdynamo it required to import openvino.torch. Possibly better to move this line to the same place where you call torch.compile for consistency and transperency of usage</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (eaidova) [inline notebooks/257-stable-diffusion-torchdynamo-backend/257-stable-diffusion-torchdynamo-backend.ipynb:1]: <p>here need some title for logical separation of diffusers pipeline initialization and description how to use torchbecked.</p><p> </p><p>Also I think variables setting also should be placed in this section for better understanding how user can control torch.compile flow</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (eaidova) [inline notebooks/257-stable-diffusion-torchdynamo-backend/257-stable-diffusion-torchdynamo-backend.ipynb:1]: <p>please align gradio with recommendations from contribution guide</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Comment by eaidova** [comment]
@likholat thanks for notebook. I have 2 questions:
1. Is it principal  that you use stable-diffusion-1.5? Maybe better to use stable-diffusion-2.1? We have collection of notebooks for this model describing different ways how to run it (just native conversion, optimum-intel), I think it can be useful to continue this notebooks family with torch.compile that users can compare all available ways to run models.
2. I think it is also great opportunity to mention that using this approach stable diffusion works in stable-diffuison-webui with help of torch compile and we can use notebook for advertising this apporach. what do you think?

**Comment by likholat** [comment]
@eaidova thanks for comments, I'll update the notebook

**Comment by likholat** [comment]
@eaidova please, take a look

File: notebooks/263-stable-diffusion-torchdynamo-backend/263-stable-diffusion-torchdynamo-backend.ipynb [hunk 2]
```diff
@@ -0,0 +1,420 @@
+{
```
Comment (eaidova) [inline notebooks/263-stable-diffusion-torchdynamo-backend/263-stable-diffusion-torchdynamo-backend.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    %pip install -q "torch==2.0.1"
</pre><p>does torch 2.1 not supported?</p><p>we prefer do not use strict dependencies if there is no reason for that</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (eaidova) [inline notebooks/263-stable-diffusion-torchdynamo-backend/263-stable-diffusion-torchdynamo-backend.ipynb:1]: <p>having 2 pipelines in memory in the same time can requires extra RAM. Maybe we can split demonstration that allow users who have less RAM successfully try notebook (free memory after first pipeline usage, then run the second) and also allow us highlight that the same actions required for both tasks.</p><p> </p><p>When we run ov directly we combain both pipelines in one and reuse memory running both task, but with torch.compile it is not applicable</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (eaidova) [inline notebooks/263-stable-diffusion-torchdynamo-backend/263-stable-diffusion-torchdynamo-backend.ipynb:1]: <p> I believe these functions required only for gradio? There is no need any extra wrappers for running pipeline, right?</p><p>I think it is important point to show how simple to use torch.compile that nothing changed from original pipeline call and only one line of code added. </p><p>Using wrappers for call pipeline can may weaken this effect and reduce readability.</p><p> </p><p>If it is used only for gradio, can it be moved closed to gradio and use pipeline directly for the places?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (eaidova) [inline notebooks/263-stable-diffusion-torchdynamo-backend/263-stable-diffusion-torchdynamo-backend.ipynb:1]: <p>maybe just Run Image generation?</p><p>current title sounds unclear like you do not know what exact you run</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1338/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Comment by likholat** [comment]
@eaidova please, take a look

**Comment by likholat** [comment]
@eaidova I'll add [`options` param](https://github.com/openvinotoolkit/openvino/pull/20946) instead of env variables after the upcoming release

**Comment by eaidova** [comment]
> @eaidova I'll add [`options` param](https://github.com/openvinotoolkit/openvino/pull/20946) instead of env variables after the upcoming release

@likholat you can add it rigfht now installing openvino-nightly instead of openvino package (we will update requirements automatically, when release happens)

**Comment by likholat** [comment]
> > @eaidova I'll add [`options` param](https://github.com/openvinotoolkit/openvino/pull/20946) instead of env variables after the upcoming release
> 
> @likholat you can add it rigfht now installing openvino-nightly instead of openvino package (we will update requirements automatically, when release happens)

Done

**Review by eaidova — APPROVED** [review]

**Comment by eaidova** [comment]
@likholat great work! Thank you!

**Pull request merged**